tests/core.py

```diff
@@ -166,6 +176,7 @@ def test_schedule_dag_relativedelta(self):
             task_id="faketastic",
             owner='Also fake',
             start_date=datetime(2015, 1, 2, 0, 0)))
+        dag.sync_to_db()
```

Comment:
Maybe a dumb question, but it's not clear to me why we need all of these new calls to `sync_to_db()` scattered throughout the tests (or how you identified all of the spots where we need to sync). Can you elaborate on the impact of *not* explicitly syncing the DagModel?

Reply:
This is because a lot of tests are working to a specific state of the database. Because I'm now actively using the DagModel table in the `create_dagrun()` method is was needed to add to the database in a lot of places. This was mainly trial and error on travis, adding it to tests that does fail inside `create_dagrun()`.

Doing `sync_do_db()` is a lot easier then manual adding all of the DagModel records for each test.